lib/gameboard/board.rb

```diff
@@ -0,0 +1,517 @@
+    #
+    #   full?
+    #     #=> false (because our default piece is now "Y")
+    #
+    def full?
+      board.none? { |cell| cell.value == cells  }
+    end
+
+    # Public: Return every row on the gameboard.
+    #
+    # coords: An optional boolean to specify returning an Array of coordinates
+    #         instead of values.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   horizontal
+    #     #=> [
+    #          ["X", "Y", "Z"],
+    #          ["A", "B", "C"],
+    #          [1, 2, 3]
+    #         ]
+    #
+    #   horizontal(true)
+    #     #=> [
+    #          [[0, 0], [1, 0], [2, 0]],
+    #          [[0, 1], [1, 1], [2, 1]],
+    #          [[0, 2], [1, 2], [2, 2]]]
+    #         ]
+    #
+    def horizontal(coords = false)
+      rows = []
+      height.times do |y|
+        rows << board.select { |cell| cell.coord.y == y }.map do |cell|
+          coords ? cell.coord.position : cell.value
+        end
+      end
+      rows
+    end
+
+    # Public: Nail the board down so it can't be flipped.
+    def nail_down
+      @nailed = true
+    end
+
+    # Public: Return whether the board is nailed down.
+    def nailed_down?
+      !!nailed
+    end
+
+    # Public: Return all valid neighbors of a given coordinate.
+    #
+    # point  - The X,Y coordinate neighbors should be based off of.
+    # coords - An optional boolean to return an array of coordinate
+    #          pairs
+    #
+    #
+    # Examples
+    #                   -------------
+    #                 0 | 1 | 2 | 3 |
+    #                   -------------
+    #                 1 | A | B | C |
+    #                   -------------
+    #                 2 | X | Y | Z |
+    #                   -------------
+    #                   - 0 - 1 - 2 -
+    #
+    #   neighbors([1,1])
+    #           #=> [1, 2, 3, "A", "C", "X", "A", 1]
+    #
+    #   neighbors([1,1] true)
+    #           #=> [
+    #                [0, 2], [1, 2],
+    #                [2, 2], [0, 1],
+    #                [2, 1], [0, 0],
+    #                [0, 1], [0, 2]
+    #               ]
+    #
+    #
+    def neighbors(point, coords = false)
+      temp = Coordinate.new(point[0], point[1])
+      valid_neighbors(temp).map do |cell|
+        coords ? cell.coord.position : cell.value
+      end
+    end
+
+    # Public: Set random cells to a given value.
+    #
+    # piece - A value to set to cells
+    #
+    def randomize(piece)
+      rand(1...board.length).times do
+        board[rand(board.length)].value = piece
+      end
+    end
+    # Public: Set a cells value at a given coordinate
+    #
+    # coord - A coordinate pair in the form X, Y
+    def set_cell(coord, value)
+      cell = find_cell(coord)
+      raise "Cell Not Found" unless !!cell
```

Comment:
I usually only use the !! on return values, so that people can't misuse the public interface. You're safe to rely on the "truthiness" of objects here.

Also, I think Coord could be your Object here as well (and all further cases). This would allow you to easily feed values from cells back into the board within coercion.